### PR TITLE
Bullet component

### PR DESCRIPTION
**Pull request by DenBedste**

### CODE REVIEW

**Comment by DenBedste** [comment]
Check failed caused by PositionPart constructor now taking (int x, int y) opossed to (Point(x, y))

**Review by stped18 — APPROVED** [review]

**Review by Kjalris — APPROVED** [review]

File: .gitignore [hunk 1]
```diff
@@ -147,3 +147,4 @@ fabric.properties
 .idea/
 /Common/nbproject/
 /Map/nbproject/
+/dk.sdu.se4.map/nbproject/
```
Comment (Nicklason) [inline .gitignore:150]: Don't do this.

File: dk.sdu.se4.Equipment/pom.xml [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+  <parent>
+    <relativePath>../poms/compiled/</relativePath>
+    <groupId>dk.sdu.se4.semester-project.build</groupId>
+    <artifactId>compiled-bundle-settings</artifactId>
+    <version>1.0-SNAPSHOT</version>
+  </parent>
+
+  <properties>
+    <bundle.symbolicName>dk.sdu.se4.Equipment</bundle.symbolicName>
+    <bundle.namespace>dk.sdu.se4.Equipment</bundle.namespace>
```
Comment (Nicklason) [inline dk.sdu.se4.Equipment/pom.xml:13]: Use lowercase name

File: dk.sdu.se4.Equipment/pom.xml [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+  </properties>
+
+  <modelVersion>4.0.0</modelVersion>
+  <groupId>dk.sdu.se4.semester-project</groupId>
+  <artifactId>dk.sdu.se4.Equipment</artifactId>
```
Comment (Nicklason) [inline dk.sdu.se4.Equipment/pom.xml:18]: Use lowercase name

File: dk.sdu.se4.Equipment/src/main/java/dk/sdu/se4/Equipment/Bullet.java [hunk 4]
```diff
@@ -0,0 +1,16 @@
+/*
+ * To change this license header, choose License Headers in Project Properties.
+ * To change this template file, choose Tools | Templates
+ * and open the template in the editor.
+ */
+package dk.sdu.se4.Equipment;
```
Comment (Nicklason) [inline dk.sdu.se4.Equipment/src/main/java/dk/sdu/se4/Equipment/Bullet.java:6]: Use lowercase name

File: dk.sdu.se4.Equipment/src/main/java/dk/sdu/se4/Equipment/BulletPlugin.java [hunk 5]
```diff
@@ -0,0 +1,53 @@
+package dk.sdu.se4.Equipment;
```
Comment (Nicklason) [inline dk.sdu.se4.Equipment/src/main/java/dk/sdu/se4/Equipment/BulletPlugin.java:1]: Use lowercase name

File: dk.sdu.se4.Equipment/src/main/java/dk/sdu/se4/Equipment/BulletPlugin.java [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+    private Entity bullet;
+
+    @Override
+    public void load() {
+         if(this.mapService != null) {
+             bullet = create();
+             this.mapService.addEntity(bullet);
+             System.out.println(bullet.toString());
+        } else {
+            System.out.println("mapservice null load");
```
Comment (Nicklason) [inline dk.sdu.se4.Equipment/src/main/java/dk/sdu/se4/Equipment/BulletPlugin.java:22]: Remove print

File: dk.sdu.se4.Equipment/src/main/java/dk/sdu/se4/Equipment/BulletPlugin.java [hunk 7]
```diff
@@ -0,0 +1,53 @@
...
+    public void unload() {
+        if(this.mapService != null) {
+            for(Entity e : this.mapService.getEntities(Bullet.class)) {
+                TimePart timePart = e.getPart(TimePart.class);
+                if(timePart.getTime() == 0) {
+                    this.mapService.removeEntity(e);
+                }
+            }
+        } else {
+            System.out.println("mapservice null unload");
```
Comment (Nicklason) [inline dk.sdu.se4.Equipment/src/main/java/dk/sdu/se4/Equipment/BulletPlugin.java:36]: Remove print

File: dk.sdu.se4.Equipment/src/main/java/dk/sdu/se4/Equipment/BulletProcessor.java [hunk 8]
```diff
@@ -0,0 +1,30 @@
+package dk.sdu.se4.Equipment;
```
Comment (Nicklason) [inline dk.sdu.se4.Equipment/src/main/java/dk/sdu/se4/Equipment/BulletProcessor.java:1]: Use lowercase name

File: dk.sdu.se4.Equipment/src/main/java/dk/sdu/se4/Equipment/BulletProcessor.java [hunk 9]
```diff
@@ -0,0 +1,30 @@
...
+        if(this.mapService != null) {
+            for(Entity e : this.mapService.getEntities(Bullet.class)) {
+                PositionPart positionPart = e.getPart(PositionPart.class);
+                MovingPart movingPart = e.getPart(MovingPart.class);
+                TimePart timePart = e.getPart(TimePart.class);
+            
+                movingPart.setMovingUp(true);
+                movingPart.setMovingDown(false);
+                movingPart.setMovingRight(false);
+                movingPart.setMovingLeft(false);
```
Comment (Nicklason) [inline dk.sdu.se4.Equipment/src/main/java/dk/sdu/se4/Equipment/BulletProcessor.java:22]: Always shoot a bullet up?

File: dk.sdu.se4.Equipment/src/main/java/dk/sdu/se4/Equipment/EquipmentCore.java [hunk 10]
```diff
@@ -0,0 +1,18 @@
+package dk.sdu.se4.Equipment;
+
+import dk.sdu.se4.common.service.MapService;
+
+public class EquipmentCore {
+    protected MapService mapService=null;
+    
+    public void addMapService(MapService mapService) {
+        this.mapService = mapService;
+        System.out.println("ADDED MapService TO Bullet "+this.mapService.getClass().toString());
```
Comment (Nicklason) [inline dk.sdu.se4.Equipment/src/main/java/dk/sdu/se4/Equipment/EquipmentCore.java:10]: Remove print

File: dk.sdu.se4.Equipment/src/main/java/dk/sdu/se4/Equipment/EquipmentCore.java [hunk 11]
```diff
@@ -0,0 +1,18 @@
...
+    }
+
+    public void removeMapService(MapService mapService) {
+        System.out.println("REMOVED MapService FROM Bullet "+this.mapService.getClass().toString());
```
Comment (Nicklason) [inline dk.sdu.se4.Equipment/src/main/java/dk/sdu/se4/Equipment/EquipmentCore.java:14]: Remove print

File: dk.sdu.se4.Equipment/src/main/resources/META-INF/EquipmentPlugin.xml [hunk 12]
```diff
@@ -0,0 +1,10 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<scr:component xmlns:scr="http://www.osgi.org/xmlns/scr/v1.1.0" name="dk.sdu.se4.Equipment.plugin">
+    <implementation class="dk.sdu.se4.Equipment.BulletPlugin"/>
```
Comment (Nicklason) [inline dk.sdu.se4.Equipment/src/main/resources/META-INF/EquipmentPlugin.xml:3]: Use lowercase name

File: dk.sdu.se4.Equipment/src/main/resources/META-INF/EquipmentProcessor.xml [hunk 13]
```diff
@@ -0,0 +1,10 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<scr:component xmlns:scr="http://www.osgi.org/xmlns/scr/v1.1.0" name="dk.sdu.se4.Equipment">
+    <implementation class="dk.sdu.se4.Equipment.BulletProcessor"/>
```
Comment (Nicklason) [inline dk.sdu.se4.Equipment/src/main/resources/META-INF/EquipmentProcessor.xml:3]: Use lowercase name

File: dk.sdu.se4.common/src/main/java/dk/sdu/se4/common/entity/part/TimePart.java [hunk 14]
```diff
@@ -0,0 +1,24 @@
+package dk.sdu.se4.common.entity.part;
+
+import dk.sdu.se4.common.entity.Entity;
+
+public class TimePart implements EntityPart {
```
Comment (Nicklason) [inline dk.sdu.se4.common/src/main/java/dk/sdu/se4/common/entity/part/TimePart.java:5]: Add comments. What is "time" and why is it a float?

File: pom.xml [hunk 15]
```diff
@@ -53,6 +53,7 @@
     <module>dk.sdu.se4.common</module>
     <module>dk.sdu.se4.map</module>
     <module>dk.sdu.se4.libgdx</module>
+    <module>dk.sdu.se4.Equipment</module>
```
Comment (Nicklason) [inline pom.xml:56]: Use lowercase name

**Review by Nicklason — CHANGES_REQUESTED** [review]

**Comment by Nicklason** [comment]
Why did you delete the test?

**Review by Nicklason — APPROVED** [review]

**Pull request merged**